jupyterlab_wipp_plugin_creator/jupyterlab_wipp_plugin_creator/handlers.py

```diff
@@ -0,0 +1,128 @@
+import json
+import os
+
+from shutil import copy2
+from wipp_client.wipp import gen_random_object_id
+
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.utils import url_path_join
+from .log import get_logger
+import tornado
+
+from jinja2 import Template
+
+logger = get_logger()
+# logger.setLevel(logging.INFO)
+
+class WippHandler(APIHandler):
+    @property
+    def wipp(self):
+        return self.settings["wipp-plugin-creator"]
+
+
+# Create wipp plugin based on user input
+class CreatePlugin(WippHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        POST request handler,
+
+        Creates a temp folder with random name,
+        Generates plugin.json, requirements.txt, dockerfile,
+        Copies files
+        Builds dockerimage,
+
+        Input format:
+            {
+                'formdata':
+                    {
+                    'name': '',
+                    'version': '',
+                    'title': '',
+                    'description': ''
+                    }
+                'filepaths':
+                    string[]
+            }
+
+        """
+
+        pwd = os.getcwd()
+
+        # Random ID follows MongoDB format
+        randomId = gen_random_object_id()
+
+        pluginOutputPath = os.getenv("PLUGIN_TEMP_LOCATION")
+        # if ENV exists
+        if pluginOutputPath:
+            os.chdir(pluginOutputPath)
+            logger.info(f"ENV variable exists, output path set to {pluginOutputPath}.")
+            os.makedirs(f"{randomId}")
+            os.chdir(f"{randomId}")
+            pluginOutputPath = os.getcwd()
+            logger.info("Random folder name created: ", pluginOutputPath)
+
+        else:
+            logger.error("ENV variable doesn't exist, please use command 'export PLUGIN_TEMP_LOCATION = '...' to set. Terminating..")
+            quit()
+
+        # Read POST request
+        data = json.loads(self.request.body.decode("utf-8"))
+        form = data["formdata"]
+        filepaths = data["addedfilepaths"]
+        requirements = form["requirements"]
+
+        # Separate requirements key in the formdata form rest to write plugin.json and requirements.txt separately
+        form.pop("requirements")
+        form["containerId"] = "polusai/generated-plugins:" + randomId
+
+        # register plugin manifest to wipp CI
+        self.wipp.register_plugin(form)
+
+        # Generate files to temp folder
+        try:
+            with open("plugin.json", "w") as f1:
+                f1.write(json.dumps(form))
+            with open("requirements.txt", "w") as f2:
+                for req in requirements:
+                    f2.write(f"{req}\n")
+            # Read from Jinja2 template
+            os.chdir(pwd + '/jupyterlab_wipp_plugin_creator')
```

Comment:
This line is a bug: it relies on the fact that there is `jupyterlab_wipp_plugin_creator` in your process folder (user's home folder). Either, find the reliable way to load the file from where the Python package is installed; *or* create a config folder which going to host the Dockerfile Jinja2 template.

Reply:
I think this a relevant method if you go the former way: https://docs.python.org/3.9/distutils/setupscript.html?highlight=data_files#installing-additional-files